data_steward/resources.py

```diff
@@ -575,9 +576,43 @@ def get_git_tag():
 
 def mapping_table_for(domain_table):
     """
-
     Get name of mapping table generated for a domain table
     :param domain_table: one of the domain tables (e.g. 'visit_occurrence', 'condition_occurrence')
     :return:
     """
     return f'_mapping_{domain_table}'
+
+
+def ext_table_for(domain_table):
+    """
+    Get name of extention table generated for a domain table
+    :param domain_table: one of the domain tables (e.g. 'visit_occurrence', 'condition_occurrence')
+    :return:
+    """
+    return f'{domain_table}_ext'
+
+
+def ask_if_continue() -> None:
+    """
+    Checks if the user wishes to continue running the script or not.
+    This function is placed after each step as a checkpoint.
+    Raises:
+        RuntimeError: Abort the execution when the user wishes not to continue.
+    """
+    confirm = input("\nContinue? [Y/N]:\n\n")
+    if confirm.upper() != 'Y':
+        raise RuntimeError('User canceled the execution.')
+
+
+def get_new_dataset_name(src_dataset_name: str, release_tag: str) -> str:
+    """
+    Based on the old dataset name and the new release tag, creates a new
+    dataset name. This function assumes both the dataset name and the release
+    tag follow the Curation team's naming convention. It may return an
+    incorrect dataset name if those are irregularly named.
+    Args:
+        src_dataset_name: Name of the source dataset.
+        release_tag: Release tag for the new datasets.
+    Returns: Name of the new dataset.
+    """
+    return re.sub(r'\d{4}q\dr\d', release_tag, src_dataset_name)
```

Comment:
Hi, @hiro-mishima - might we suggest the following?
`r'\d{4}q[1-4]r\d{1,3}'`
This will:

- ensure quarter is one digit only, 1-4
- allow 1, 2, or 3 digit revision numbers

Reply:
Thanks for your suggestion, @molly-giddings! I asked @MichaelSchmidt82 to include this change since he is working on a relevant ticket right now: https://github.com/all-of-us/curation/pull/1491#issuecomment-1428748870

Reply:
I ended up adding this suggestion to the open PR for DC-3079. Thanks for the suggestion!
https://github.com/all-of-us/curation/pull/1489/commits/4b12430e03556fc106b1435f871ffd2f94aaa35d#diff-6ecbf63270b1fe0210a08915e522c72d8d164fdb10460211699a92511fb6e929R618